packages/algorand-js/src/types.ts

```diff
@@ -95,3 +95,13 @@ export interface StoreAccount {
   getLocalState: (appId: number, key: Uint8Array) => StackElem | undefined
   updateLocalState: (appId: number, key: Uint8Array, value: StackElem) => AppLocalState[]
 }
+
+// https://developer.algorand.org/docs/reference/teal/specification/#oncomplete
+export enum TxnOnComplete {
+  NoOp,
+  OptIn,
+  CloseOut,
+  ClearState,
+  UpdateApplication,
+  DeleteApplication
```

Comment:
this `enum` will result in reverse mapping, (1 -> NoOp and NoOp -> 1), to solve this we can use `string` enum.
Ex: NoOp = '0'. 

https://www.typescriptlang.org/docs/handbook/enums.html

Reply:
Reverse mapping is already handled in the constructor. I like the idea to use string enum. But shouldn't we use a map instead then? What's the difference?

Reply:
yes, it is handled but i don't think it is good to create reverse mapping and then not use it.
I couldn't find the exact difference in typescript. Please let me know if you find it.

Reply:
after reading more about it, i found that `enum` are stored as array like structure, that's why to find key, we have to traverse all the values in enum,
But Map is implemented using hashtable. 
Therefore it is way much faster, but in our case we have  only 5 or 6 pairs.

Reply:
Reverse Mapping is a general property of any enum. When we decided on call we should use enums, it was understood that there would be reverse mapping.
I would prefer to use map instead of enums then. Please let me know if you follow, i'll update it.

Reply:
I don't think we discussed for this enum, In this case what if we have TypeEnumConstants also, then there will be two values mapped to `0` -> NoOp, unknown. But for other enums let's say TransactionType, we can use reverse mapping.

I agree, your call

Reply:
Yes, but in this case reverse mapping acutally makes sense, because for two strings we can have the same `int` as well. eg.
```
int NoOp -> int 0 // using TxnOnComplete Enum
int pay -> int 0 // using TypeEnum
```
So both these statements makes sense: 
a) `NoOp` and `pay` represent int 0, 
b) int `0` represents both `NoOp` and `pay` 

I am retaining the enum for now with number mapping.

Reply:
The thing I didin't like in the code was when we use `Object.values().indexOf(...)` (it's ugly and not intuitive), otherwise @amityadav0  has a good point - iterating over a list of few elements doesn't change much.

In general we are using this in few places, so it's good to have some shorter, better way to do it: map, auxiliary function etc...